### PR TITLE
[Client/Team4] major (game-breaking) bugfixing

### PR DESCRIPTION
**Pull request by akowal3**
- Fixed major bugs with agent implementation:
  - no negative allocation requests
  - no negative allocation taking
  - follows rules for tax/allocations
- improved foraging
- added configuration:
  - `DefaultClient(id shared.ClientID) baseclient.Client`
  - `DishonestClient(id shared.ClientID) baseclient.Client`
  - `ModerateClient(id shared.ClientID) baseclient.Client`

### CODE REVIEW

File: internal/clients/team4/forage.go [hunk 1]
```diff
@@ -16,18 +15,74 @@ type forageHistory struct {
...
+	scale := 1.4 * c.getSafeResourceLevel()
+	forageContribution := c.getResources() - (2-shared.Resources(c.internalParam.riskTaking))*scale
+	c.Logf("team4 resource level: %v", c.getResources())
+	if c.getSafeResourceLevel()*2 > c.getResources()-forageContribution {
+		forageContribution = c.getResources() - c.getSafeResourceLevel()*2
+	}
+	if forageContribution < 0 {
+		forageContribution = 0
+	}
+	if forageContribution < forageMax {
```
Comment (preetl) [inline internal/clients/team4/forage.go]: should this condition be > ?

Comment (akowal3) [reply internal/clients/team4/forage.go]: @tobhil98?

Comment (tobhil98) [reply internal/clients/team4/forage.go]: Yeah, Thanks

**Review by NGovani — APPROVED** [review]
All right you're getting the green light. Here's hopin it work

**Pull request merged**